### ISSUE TITLE
Project.move_to doesn't keep first level of hierarchy

### ISSUE THREAD

**Issue by pmrv** [pyiron/pyiron_base#480]
Let's say you have a two project and you want to move one of them
```
- src
  | - job1
  | - job2
  \ - subfoo
    | - job3
    \ - job4
- target (empty)
```
If you do `Project('src').move_to(Project('target'))` afterwards you'll have
```
- src (now empty)
- target
  | - job1
  | - job2
  \ - subfoo
    | - job3
    \ - job4
```
i.e. `move_to` behaves like `mv src/* target/`.  I'd rather like to behave like `mv src target/`, so that the final result would be
```
- src (now empty)
- target
  \ - src
    | - job1
    | - job2
    \ - subfoo
      | - job3
      \ - job4
```

Any thoughts as to which behavior is better?

I got tripped by this because I did something like
```python
for g in pr.list_groups():
  if skip_some(g): continue
  pr[g].move_to(Project(...))
```
when I really should have done
```python
for g in pr.list_groups():
  if skip_some(g): continue
  pr[g].move_to(Project(...).create_group(g))
```
so now I have all the jobs of each `pr[g]` in one project instead of their original subprojects.  It's a pain I'd like to avoid in the future, but I thought I get your thoughts first, since this changes behavior.

**Comment by pmrv** [comment]
Consensus:
- current behavior should be covered by a new method 'rename', *but* give error if target exists
- change behavior of `move_to` to described behavior

**Comment by samwaseda** [comment]
> Consensus:
> 
> * current behavior should be covered by a new method 'rename', _but_ give error if target exists
> * change behavior of `move_to` to described behavior

I cannot really remember what we said yesterday, but if this consensus is still valid, isn't it dangerous to suddenly change the behaviour of `move_to`? There might be people who are already using the current `move_to`.

**Comment by pmrv** [comment]
Yes, that's an issue.  I guess we can hide the new behavior behind a flag and issue a deprecation warning if the flag is not passed.
